spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/procedures/RewriteDataFilesProcedure.java

```diff
@@ -107,10 +122,22 @@ public InternalRow[] call(InternalRow args) {
     String sortOrderString = input.asString(SORT_ORDER_PARAM, null);
     Map<String, String> options = input.asStringMap(OPTIONS_PARAM, ImmutableMap.of());
     String where = input.asString(WHERE_PARAM, null);
+    String partitionedByString = input.asString(PARTITIONED_BY_PARAM, null);
+    Boolean createPartitionIfNotExists =
+        input.asBoolean(CREATE_PARTITION_IF_NOT_EXISTS_PARAM, false);
 
     return modifyIcebergTable(
         tableIdent,
         table -> {
+          if (partitionedByString != null || createPartitionIfNotExists) {
+            Integer partitionSpecId =
+                checkAndPreparePartitionSpec(
+                    table, partitionedByString, createPartitionIfNotExists, options);
+            options.put(OUTPUT_SPEC_ID, partitionSpecId.toString());
```

Comment:
nitpick: I guess I could set this inside of the `checkAndPreparePartitionSpec` method